src/utils.rs

```diff
@@ -42,6 +42,12 @@ impl PartialEq for EngineVersion {
     }
 }
 
+impl PartialOrd for EngineVersion {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some((*self as u8).cmp(&(*other as u8)))
+    }
+}
```

Comment:
you can just do
`#[derive(PartialOrd, PartialEq, Eq, Debug, Copy, Clone)]`
on `EngineVersion`

Reply:
I tried this before and it didn't work. But now, hilariously, it's working. I will fix